XenAdmin/Dialogs/RepairSRDialog.cs

```diff
@@ -282,7 +283,8 @@ private void RepairButton_Click(object sender, EventArgs e)
             
             _repairAction.Changed += action_Changed;
             _repairAction.Completed += action_Completed;
-            Grow(_repairAction.RunAsync);
+            if(!isModalMode)

```

Comment:
The two lines above should also be inside the `if` block because these event handlers will try to update the controls on a dialog that has been closed.